sdk/tables/Azure.Data.Tables/src/DynamicTableEntity.cs

```diff
@@ -0,0 +1,221 @@
+        /// <value>A <see cref="DateTimeOffset"/> containing the timestamp of the entity.</value>
+        public DateTimeOffset Timestamp
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.TimeStamp, out object value) ? (DateTimeOffset)value : default; }
+        }
+
+        /// <summary>
+        /// Gets or sets the entity's ETag. Set this value to '*' in order to force an overwrite to an entity as part of an update operation.
+        /// </summary>
+        /// <value>A string containing the ETag value for the entity.</value>
+        public string ETag
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.Etag, out object value) ? (string)value : default; }
+            set { _properties[TableConstants.PropertyNames.Etag] = value; }
+        }
+
+        /// <summary>
+        /// Constructs an instance of a <see cref="TableEntity" />.
+        /// </summary>
+        public DynamicTableEntity()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicTableEntity"/> class with the specified partition key and row key.
+        /// </summary>
+        /// <param name="partitionKey">A string containing the partition key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        /// <param name="rowKey">A string containing the row key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        public DynamicTableEntity(string partitionKey, string rowKey)
+        {
+            PartitionKey = partitionKey;
+            RowKey = rowKey;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicTableEntity"/> class with properties in the <see cref="IDictionary"/>.
+        /// </summary>
+        /// <param name="values">A <see cref="IDictionary"/> containing the initial values of the entity.</param>
+        public DynamicTableEntity(IDictionary<string, object> values)
+        {
+            _properties = values != null ?
+                new Dictionary<string, object>(values) :
+                new Dictionary<string, object>();
+        }
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="String"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public string GetString(string key) => GetValue<string>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="byte"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public byte[] GetBinary(string key) => GetValue<byte[]>(key);
```

Comment:
`Binary` is interesting...  Should this use the new `BinaryData` type instead of `byte[]`?

Reply:
Could you expand on what benefit you see in making that change? Is it to provide some extra serialization help to user's of `byte[]`?

Reply:
Do users want that as `byte[]`, `ReadOnlyMemory<byte>`, `Stream`?  Or is this field representing an encoded string with UTF-8 or UTF-16?  Or maybe even a JSON document they've serialized?  `BinaryData` wraps all of that.

Reply:
(And to be clear, this can happen in the future.  I think `byte[]` is fine for now.)

Reply:
tracking this in #13640